### ISSUE TITLE
Required field in form input not stored correctly

### ISSUE THREAD

**Issue by firepinn** [madoublet/respond#397]
When setting the required field in a form field to Yes in the latest git version of respondcms (e7ff5081a4634cc3a24fe0a376b14e833646f23e), after publishing and reloading the page, the required parameter is set to No again. Consequently, validation in the corresponding forms doesn't work.

**Comment by axis80** [comment]
I've confirmed that this is an issue.  I've made a little bit of progress on a fix, and am hopeful that someone smarter can help me take it the rest of the way.  Here are my findings thus far:

**Observation 1**

The form element does have a "generate" method, which is defined on line 561 of /plugins/component/respond.component.js.  

From what I can tell, the generate method is called when the user clicks "Save and Publish".  It is called on each element to render it into an HTML fragment which is then stored in the db.  

However, the generate method mentioned above is not being called when the user clicks "Save and Publish".  I added debugging code near line 1088 of /js/respond.editor.js.  Here is the relevant portion of code:

    // get index from the menu
    var i = utilities.getIndexByAttribute(respond.editor.menu, 'class', cssClass);
    
    // execute the generate method for the plugin
    if(i != -1){

Questions for smart people:

Is the generate method in respond.component.js supposed to be called for the form element?

If so, then what is the purpose of the above call to getIndexByAttribute and why is it failing?


**Observation 2**

In /plugins/component/respond.component.js (line 516) is this code which parses an existing form and loads it into the editor:

    if(required == 'required'){
        required = 'false';
    }

That seems backwards to me.  I think it should be as follows, but I'm not going to commit the change until I have resolved item 1 above.

    if(required == 'required'){
        required = 'true';
    } else {
        required = 'false';
    }
